lib/sig_item_map.ml

```diff
@@ -40,8 +40,12 @@ let add (type a) ~name (item_type : a item_type) (item : a)
         types_map = String_map.add name item types_map;
       }
 
-let diff (type a)
-    ~(diff_item : a item_type -> string -> a option -> a option -> 'diff option)
+type 'diff poly_diff_item = {
+  diff_item :
+    'a. 'a item_type -> string -> 'a option -> 'a option -> 'diff option;
```

Comment:
This type is long and written in several places.

Let's expose it as a (parametrized) type alias, to avoid duplication and make things more readable!

Reply:
What I mean is to define `type ('a, 'diff) diff_item = 'a item_type -> string -> 'a option -> 'a option -> 'diff option` first, expose it in the interface, and use it anywhere needed (I did not give much though to the name).
In particular, this can simplify [these line](https://github.com/ocaml-semver/ocaml-api-watch/pull/98/files#diff-016e53099417551f0595b3b00365c6083886f4e4208b555bb6901bc63a438f82R108-R109).

Reply:
(A type alias is a particular type declaration, where you are not introducing anything new, just giving a name to an existing type.

When you are defining a record, that is not a type alias: you are creating a new thing.
When you do `type t = <type expression>`, the type already existed and could be used by inlining the type expression.)

Reply:
Thanks for clarifying, I misunderstood a record type as being a type alias >:<